src/Compilers/CSharp/Portable/Symbols/Source/SourceParameterSymbol.cs

```diff
@@ -23,7 +24,7 @@ internal abstract class SourceParameterSymbol : SourceParameterSymbolBase
         protected SymbolCompletionState state;
         protected readonly TypeWithAnnotations parameterType;
         private readonly string _name;
-        private readonly ImmutableArray<Location> _locations;
+        private readonly OneOrMany<Location> _locations;
```

Comment:
>private readonly OneOrMany<Location> _locations [](http://example.com/codeflow?start=8&length=47)

Is it ever many? Can we use a singleton Location?

Reply:
So, with synthesized parameters, this picks up the 'Locations' of the owner.  I would be willing to try just storing a single location and hoping that was sufficient.  But i was worried about potential error scenarios, or something going wonky doing that, so i left it OneOrMany to ensure that if there was ever 'many' that this was safe.

Let me know if you'd like me to try out a vresion with just 'one' :)

Reply:
I would do the following:
1. Add an assert that we always get one location. Never zero or more than one. 
2. Go with the singleton that is initialized to the first or NoLocation.

Reply:
Done.

Reply:
@AlekseyTs looks like getting 0 locations def happens.  As an example LocalStateTracingTests.Queries.  Would you like me to revert that last change, or try another route?  Thanks!

Reply:
> Would you like me to revert that last change, or try another route?

Perhaps we could store null to indicate the lack of location.

Reply:
I can try that today

Reply:
ok.  pushed that change to see if CI reveals anything.